test/source/dense_test.cpp

```diff
@@ -49,3 +49,32 @@ TEST_CASE("Dense-2D")
     }
     CHECK(l1 == SIZE1);
 }
+
+TEST_CASE("Dense-2D-Size")
+{
+    constexpr uintptr_t SIZE1 = 6;
+    constexpr uintptr_t SIZE2 = 6;
+    constexpr uint8_t ZERO = 0;
+    constexpr uint8_t ONE = 1;
+
+    xsparse::levels::dense<std::tuple<>, uintptr_t, uintptr_t> d1{ SIZE1 };
+    xsparse::levels::dense<std::tuple<decltype(d1)>, uintptr_t, uintptr_t> d2{ SIZE2 };
+
+
+    uintptr_t l1 = 0;
+    for (auto const [i1, p1] : d1.iter_helper(std::make_tuple(), ZERO))
+    {
+        CHECK(l1 == p1);
+        CHECK(l1 == i1);
+        uintptr_t l2 = 0;
+        for (auto const [i2, p2] : d2.iter_helper(std::make_tuple(i1), p1))
+        {
+            CHECK(l1 * d2.size(ONE) + l2 == p2);
+            CHECK(l2 == i2);
+            ++l2;
+        }
+        CHECK(l2 == d2.size(ONE));
+        ++l1;
+    }
+    CHECK(l1 == d1.size(ONE));
```

Comment:
```suggestion
    CHECK(l1 == SIZE1);
```

Reply:
That is usually needed only when iterating. You calculate the size of sized levels by passing the amount of inserted coordinates in the last level.